openwisp_monitoring/db/backends/influxdb2/client.py

```diff
@@ -0,0 +1,462 @@
+        return timestamp
+
+    def write(self, name, values, **kwargs):
+        timestamp = self._get_timestamp(timestamp=kwargs.get('timestamp'))
+        try:   
+            point = {
+                'measurement': name,
+                'tags': kwargs.get('tags'),
+                'fields': values,
+                'time': timestamp,
+            }
+            print(f"Writing point to InfluxDB: {point}")
+            self.write_api.write(bucket=self.bucket, org=self.org, record=point)
+            print("Successfully wrote point to InfluxDB")
+        except Exception as e:
+            print(f"Error writing to InfluxDB: {e}")
+
+    def batch_write(self, metric_data):
+        points = []
+        for data in metric_data:
+            timestamp = self._get_timestamp(timestamp=data.get('timestamp'))
+            point = Point(data.get('name')).tag(**data.get('tags', {})).field(**data.get('values')).time(timestamp, WritePrecision.NS)
+            points.append(point)   
+        try:
+            self.write_api.write(bucket=self.bucket, org=self.org, record=points)
+        except Exception as e:
+            logger.error(f"Error writing batch to InfluxDB: {e}")
+
+    def _format_date(self, date_str):
+        if date_str is None or date_str == 'now()':
+            return date_str
+        try:
+            date = datetime.strptime(date_str, '%Y-%m-%d %H:%M:%S')
+            return date.strftime('%Y-%m-%dT%H:%M:%SZ')
+        except ValueError:
+            # If the date_str is not in the expected format, return it as is
+            return date_str
+        
+    def query(self, query):
+        print(f"Executing query: {query}")
+        try:
+            result = self.query_api.query(query)
+            print(f"Query result: {result}")
+            return result
+        except Exception as e:
+            print(f"Error executing query: {e}")
+            logger.error(f"Error executing query: {e}")
+            return []
+        
+    def _parse_query_result(self, result):
+        print("Parsing query result")
+        parsed_result = []
+        for table in result:
+            for record in table.records:
+                parsed_record = {
+                    'time': record.get_time().isoformat(),
+                }
+                for key, value in record.values.items():
+                    if key not in ['_time', '_start', '_stop', '_measurement']:
+                        parsed_record[key] = value
+                parsed_result.append(parsed_record)
+        print(f"Parsed result: {parsed_result}")
+        return parsed_result
+ 
+    def read(self, key, fields, tags, **kwargs):
+        extra_fields = kwargs.get('extra_fields')
+        since = kwargs.get('since', '-30d')  # Default to last 30 days if not specified
+        order = kwargs.get('order')
+        limit = kwargs.get('limit')
+        bucket = self.bucket
+
+        # Start building the Flux query
+        flux_query = f'from(bucket:"{bucket}")'
```

Comment:
@praptisharma28 why are you building the query like this here? Why aren't you using the queries defined in queries.py? 

This looks wrong. Charts for different metrics are handled differently. That's why there are individual queries for each chart.

Reply:
My bad, I got confused between the  `get_list_queries` and `read` method. The current implementation is okay.